src/components/header/index.js

```diff
@@ -50,10 +53,53 @@ class Header extends PureComponent {
   };
 
   componentDidMount() {
-    if (!this.props.pathname && typeof window !== 'undefined') {
-      this.setState({ pathname: window.location.pathname });
+    this.checkLoggedIn();
+    this.getLanguages();
+    this.findPathname();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const { relative } = this.props;
+    const { showSubmenu } = this.state;
+    if (!relative && prevState.showSubmenu && !showSubmenu) {
+      document.body.classList.remove('Header__no-scroll');
+    } else if (!relative && !prevState.showSubmenu && showSubmenu) {
+      document.body.classList.add('Header__no-scroll');
+    }
+  }
+
+  checkLoggedIn = () => {
+    const query = typeof window !== 'undefined' &&
+      qs.parse(window.location.search) ||
+      {};
+    const urlToken = query && query.token;
+    const token = urlToken || localStorage.getItem('userToken');
+    this.setState({ loggedIn: false, loggingIn: true });
+
+    if (urlToken) {
+      delete query.token;
+      const cleanQuery = query && qs.stringify(query);
+      window.history.pushState(
+        {},
+        '',
+        `${window.location.pathname}${cleanQuery ? `?${cleanQuery}` : ''}`
+      );
     }
 
+    checkLoggedIn(token)
```

Comment:
now, even if there's no token we're sending the request so we could add an `if (token)` here

Reply:
ah, `checkLoggedIn` has this condition already

Reply:
Good spot. Indeed it should only be done if token exists.

Reply:
Updated